src/Package.cc

```diff
@@ -1963,6 +1964,53 @@ YCPValue PkgFunctions::GetSolverFlags()
     return ret;
 }
 
+/**
+ * @builtin SetAdditionalVendors
+ *
+ * @short set additional vendors which are compatible
+ * @description
+ * Select additional compatible vendors for installation.
+ * @param list<string> List of additional vondor strings
+ * @return void
+ * @usage Pkg::SetAdditionalLocales(["openSUSE","SUSE LLC"]);
+ */
+YCPValue
+PkgFunctions::SetAdditionalVendors (const YCPList &vendorycplist)
+{
+    int i = 0;
+    zypp::VendorAttr::VendorList vendors;
+    while (i < vendorycplist->size())
+    {
+	if (vendorycplist->value(i)->isString())
+	{
+	  vendors.push_back(vendorycplist->value(i)->asString()->value());
+	}
+	else
+	{
+	    y2error("Pkg::SetAdditionalVendors ([...,%s,...]) not string", vendorycplist->value(i)->toString().c_str());
+	}
+	i++;
+    }
+
+    try
+    {
+      if ( zypp::getZYpp()->getTarget() ) {
```

Comment:
Use the [`zypp_ptr()`](https://github.com/yast/yast-pkg-bindings/blob/fd7d8b92c52baf7bedca5c20e6adfbba28f24d78/src/PkgFunctions.cc#L87) wrapper instead of calling `zypp::getZYpp()` directly.

```suggestion
      if ( zypp_ptr()->getTarget() ) {
```